### ISSUE TITLE
(Bug): Changes in Project Settings do not prompt "Save Project" on Exit or Reload

### ISSUE THREAD

**Issue by MartinLevi** [NatronGitHub/Natron#677]
### 

- [X] I'm using the latest version of Natron (not required but recommended)
- [X] I've restarted Natron and the issue persists
- [X] I've run Natron via the [command line](https://natron.readthedocs.io/en/rb-2.4/devel/natronexecution.html) and the issue persists
- [X] I've followed the [contributing guidelines](https://github.com/NatronGitHub/Natron/blob/RB-2.4/CONTRIBUTING.md) to the best of my understanding
- [X] My issue is not on the [issue tracker](https://github.com/NatronGitHub/Natron/issues?q=is%3Aissue+type%3Abug) and is not a duplicate of a forum thread
- [X] This bug is reproducible
- [X] This issue is not a feature request or a pull request

### Natron version

Natron 2.4.1 Snapshot (also happens on previous versions)

### Operating system

Windows 10 Version 21H1

### System specs

```markdown
RAM: 31.9GB
CPU: Intel(R) Core(TM) i5-8400 CPU @ 2.80GHz
No GPU
```

### Did you install Natron using the official installer?

- [X] Yes, I used the official installer
- [ ] No, I installed from a binary archive
- [ ] No, I compiled Natron from sources
- [ ] No, I installed Natron via another method

### What were you trying to do?

```markdown
I changed the Frame Range in my project Settings.  I exited the project, and was not prompted to save my changes.  The same issue occurs with changes to any of the Settings parameters, either on Exit from the project or on Reload.
```

### What did you expect to happen? What happened instead?

```markdown
I was expecting to be prompted to save the project changes.

On a similar (but opposite) note, if I make no changes at all but render my project out to file I am prompted to save project changes.
```

### Step-by-step reproduction instructions

```markdown
1 Open an existing Natron project
2 Open the Settings panel
3 Make a change to one of the parameters
4 Click on Exit Project or Reload Project.
```

**Comment by rodlie** [comment]
Will take a look (after a quick look it does not seem like changes to project settings trigger anything), but ...

The check for changes in a project is a bit weird, it's based on auto save it seems. I can open a project, add a node and change existing node and if I'm fast enough Natron does not detect this and will not prompt me to save on quit.

https://user-images.githubusercontent.com/34516798/136595761-a5fbc4c3-2761-43e1-8e7d-afc570febe8c.mp4

If I wait a while (for the auto save) then I get the prompt:

https://user-images.githubusercontent.com/34516798/136595797-e0e8d395-93e6-48e9-b914-21d21d2ebd60.mp4

A bit off-topic, but related.

So, changes to project settings will need to trigger a new auto save to mark the project as "dirty" is my guess.

**Comment by MartinLevi** [comment]
> The check for changes in a project is a bit weird, it's based on auto save it seems. I can open a project, add a node and change existing node and if I'm fast enough Natron does not detect this and will not prompt me to save on quit.

Yes, that definitely seems a bit weird!

**Comment by MartinLevi** [comment]
> So, changes to project settings will need to trigger a new auto save to mark the project as "dirty" is my guess.

Sounds good to me.

**Comment by rodlie** [comment]
fixed in https://github.com/NatronGitHub/Natron/commit/b68ed0874915c260d833d73ae1ed93de4ccb1c42

**Comment by MartinLevi** [comment]
> fixed in [b68ed08](https://github.com/NatronGitHub/Natron/commit/b68ed0874915c260d833d73ae1ed93de4ccb1c42)

Awesome!
